application/controllers/SurveyAdministrationController.php

```diff
@@ -813,18 +813,18 @@ public function actionChangeMultipleSurveyGroup()
 
         foreach ($aSIDs as $iSurveyID) {
             $oSurvey = Survey::model()->findByPk((int)$iSurveyID);
-            $oSurvey->gsid = $iSurveyGroupId;
```

Comment:
Why moving this line? Intended?

Reply:
It's **the** issue if you set the gid before testing is user have Permission : it was tested with new gid. If user have Survey permission onnew gid : return true.

It must be tested with current gid

Reply:
See : https://github.com/LimeSurvey/LimeSurvey/blob/d3fb278e57e5561a62b1e05c619db5b1c5fbef73/application/models/Survey.php#L2279

return with new gid here : https://github.com/LimeSurvey/LimeSurvey/blob/d3fb278e57e5561a62b1e05c619db5b1c5fbef73/application/models/Survey.php#L2294

Reply:
https://github.com/LimeSurvey/LimeSurvey/pull/3547/commits/173c2d0755b278cace63511ce9e9bad03810a270

Reply:
> Why after ? We already check if user have permission to view surveygroup in Survey model.

Ok, didn't recall it.